flow/src/main/java/com/sequenceiq/flow/service/FlowService.java

```diff
@@ -195,6 +201,29 @@ public FlowCheckResponse getFlowState(String flowId) {
         flowCheckResponse.setFlowId(flowId);
         flowCheckResponse.setHasActiveFlow(!completed("Flow", flowId, List.of(), allByFlowIdOrderByCreatedDesc));
         flowCheckResponse.setLatestFlowFinalizedAndFailed(isFlowInFailedState(allByFlowIdOrderByCreatedDesc, failHandledEvents));
+        // Get the latest flow information that will be used for error logging.
+        if (flowCheckResponse.getHasActiveFlow()) {
+            flowCheckResponse = getLatestKnownFlowCheckResponse(allByFlowIdOrderByCreatedDesc, flowCheckResponse, "flow", flowId);
+        }
+        return flowCheckResponse;
+    }
+
+    public FlowCheckResponse getLatestKnownFlowCheckResponse(List<FlowLogWithoutPayload> flowLogs, FlowCheckResponse flowCheckResponse, String type, String id) {
+        // currentState can not be null, just checking in case.
+        if (flowLogs.size() > 0 && flowLogs.get(0).getCurrentState() != null) {
+            LOGGER.info("Getting the latest known flow response using {} {}", type, id);
+            flowCheckResponse.setCurrentState(flowLogs.get(0).getCurrentState());
+            if (flowLogs.get(0).getNextEvent() != null) {
+                flowCheckResponse.setNextEvent(flowLogs.get(0).getNextEvent());
+            } else {
+                flowCheckResponse.setNextEvent(null);
+            }
+            if (flowLogs.get(0).getFlowType() != null) {
```

Comment:
Thank you @lturcsanyi ! I removed the if statement but keep the one to check getFlowType() because if it is null we are not be able to get its String version using getName(). Please let me know if you have any questions!

Reply:
The `if` is okay, but please remove the `else` branch, it is unnecessary.

Reply:
@lturcsanyi Oh you are right. I misunderstood it. Updated that. Thank you!!